controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -184,6 +172,20 @@ func CreateReconciliationPlan(instance *dsc.DataScienceCluster) *profiles.Reconc
 	profiles.ProfileConfigs = profiles.SetDefaultProfiles()
 	// Create plan for component deployment
 	profiles.PopulatePlan(profiles.ProfileConfigs[profile], plan, instance)
-	// Similarly set other profiles
+
+	// warn of odd profile combinations
+	componentName := ""
+	switch profile {
+	case profiles.ProfileServing:
+		componentName = modelmeshserving.ComponentName
+	case profiles.ProfileTraining:
+		componentName = datasciencepipelines.ComponentName
+	case profiles.ProfileWorkbench:
+		componentName = workbenches.ComponentName
+	}
+	if componentName != "" && !plan.Components[componentName] {
+		r.Log.Info("warning: profile '%v' has been selected, but component %s has been explicitly disabled in the CR", profile, componentName)
```

Comment:
I think this should be `r.Log.Infof`

Reply:
Fixed and tested. The following CR:

```yaml
apiVersion: datasciencecluster.opendatahub.io/v1alpha1
kind: DataScienceCluster
metadata:
  name: default
spec:
  components:
    datasciencepipelines:
      enabled: false
    modelmeshserving:
      enabled: true
  profile: training
```

generates the following log entry:

```log
2023-07-11T19:12:53Z INFO controllers.DataScienceCluster warning: profile has been selected, but a key component has been explicitly disabled {"profile": "training", "component name": "data-science-pipelines-operator"}
```